java/bulk-import/bulk-import-runner/src/main/java/sleeper/bulkimport/job/runner/rdd/SingleFileWritingIterator.java

```diff
@@ -100,12 +99,13 @@ public Row next() {
             return null;
         }
         try {
+            String partitionId = null;
             while (input.hasNext()) {
                 Row row = input.next();
                 if (null == parquetWriter) {
                     startTime = Instant.now();
-                    initialiseState(partitionId);
                     partitionId = getPartitionId(row);
+                    initialiseState(partitionId);
```

Comment:
Please could you add a test to SingleFileWritingIteratorTest which reproduces the failure? It looks like what's missing is an assertion of the values output by the iterator.

We can assert that it only outputs one row, and that row has the partition ID, the path to a file which is constructed from the temporary directory and the partition ID, and the number of records in the file.

It might help to fix the partition ID for your test, rather than generating a random one like the existing tests. You could use PartitionsBuilder (which is just for tests) instead of SingleFileWritingIteratorTest.getPartitionTree. You can find examples in PartitionsBuilderTest.

Reply:
Actually we can split that out as a separate issue.